packages/whenable/src/module.js

```diff
@@ -0,0 +1,24 @@
+// @ts-check
+import { prepareWhen } from './when.js';
+import { prepareWhenableKits } from './whenable.js';
+import { prepareWatch } from './watch.js';
+
+/**
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {object} [powers]
+ * @param {(reason: any) => boolean} [powers.rejectionMeansRetry]
+ * @param {(p: PromiseLike<any>, watcher: import('./watch.js').PromiseWatcher, ...args: unknown[]) => void} [powers.watchPromise]
```

Comment:
I see consistent preparation for a `watchPromise` function to have extra args after the promise-watcher. Do you have some particular use in mind for these extra args?

Reply:
Those arguments are part of the existing `VatData.watchPromise` implementation, and that function appends them to arguments of both `onFulfilled` and `onRejected`.  I don't want to commit to those semantics for just the `watch` function, but I need the `watchPromise` shim to be faithful.

Reply:
Ok, no change suggested. But do we use, or have we used, those extra arguments for anything? Was there originally some motivation in mind?

Reply:
I'm still curious about this. But this PR doesn't depend on satisfying my curiosity, so resolving.

Reply:
Only in toy examples, where a Watcher could be an object whose `onFulfilled` and `onRejected` changed its behaviour based on the additional arguments.  That would allow you to reuse the same object in different `watchPromise` call sites and specify the additional arguments to trigger those different behaviours.